ouroboros-consensus/demo-playground/TxSubmission.hs

```diff
@@ -73,8 +104,126 @@ command' c descr p =
   Main logic
 -------------------------------------------------------------------------------}
 
-handleTxSubmission :: TopologyInfo -> Mock.Tx -> IO ()
-handleTxSubmission _ _ = fail "TODO: redo handleTxSubmission"
+handleTxSubmission :: forall blk.
+                      RunDemo blk
+                   => DemoProtocol blk
+                   -> TopologyInfo
+                   -> Mock.Tx
+                   -> IO ()
+handleTxSubmission p tinfo mocktx = do
+    topoE <- readTopologyFile (topologyFile tinfo)
+    case topoE of
+         Left e  -> error e
+         Right t@(NetworkTopology nodeSetups) -> do
+
+             case M.lookup (node tinfo) (toNetworkMap t) of
```

Comment:
Maybe `member` or `notMember` would make the intent clearer: just checking presence, we don't need the actual value associated with the key.

Reply:
agree